custom_components/hubspace/light.py

```diff
@@ -114,176 +123,196 @@ def _add_entity(entities, hs, model, deviceClass, friendlyName, debug):
     return entities
 
 
-def setup_platform(
-    hass: HomeAssistant,
-    config: ConfigType,
-    add_entities: AddEntitiesCallback,
-    discovery_info: DiscoveryInfoType | None = None,
-) -> None:
-    """Set up the Awesome Light platform."""
+def manual_discovery(hs: HubSpace, friendly_names: list[str], room_names: list[str], debug: bool) -> list:
```

Comment:
Ideally this function would grab the results of auto-discovery and use the devices that match friendlyName or roomName. This would reduce the amount of discovery code required to add entities to HomeAssistant.